### PR TITLE
Quality reinforcement (8): fix unstable test cases

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

- [x] sequence_sharding: [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7834/pipeline), lance6716, https://github.com/pingcap/dm/pull/1351/commits/e579c23fe46d1bce26e9207b856c407af78781ce
- [ ] dm-master starting over 10 seconds [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7861/pipeline)
- [x] many_tables: should see error and early fail [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7881/pipeline/81), lance6716, https://github.com/pingcap/dm/commit/2d368958c7cbd70187e5567fece5d2d60c1e96b0
- [x] shardddl3 (relay?): DATA RACE [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7976/pipeline), lance6716, https://github.com/pingcap/dm/pull/1351/commits/a68cf9125cf055bbb5e186100a11f7e5b0a4d0a4
- [ ] shardddl3 DM076 [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7979/pipeline)
- [ ] some unit test using etcd [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7989/pipeline/72)

### CODE REVIEW

File: tests/sequence_sharding/run.sh [hunk 1]
```diff
@@ -50,9 +50,11 @@ function run() {
     check_log_contains $WORK_DIR/worker1/log/dm-worker.log "task can't auto resume"
 
     # resume manually
+    # this operation may not return 3 `"result": true`, because worker may
+    # - response too slowly to resume, so resume still see old error and waitOperationOk will return early
```
Comment (lance6716) [inline tests/sequence_sharding/run.sh:54]: that's not a "declarative command", and I think the right way of `waitOperationOk` of resuming is resuming at least once and see that result. Our persistent data in etcd may not support this change

File: relay/relay.go [hunk 2]
```diff
@@ -496,7 +496,7 @@ func (r *Relay) handleEvents(ctx context.Context, reader2 reader.Reader, transfo
 		}
 
 		if needSavePos {
-			err = r.SaveMeta(lastPos, lastGTID)
+			err = r.SaveMeta(lastPos, lastGTID.Clone())
```
Comment (GMHDBJD) [inline relay/relay.go:499]: Possible performance regression. But I have no idea.

Comment (lance6716) [reply relay/relay.go:499]: How about we merging this PR now? I met this DATA WARN again

**Comment by lance6716** [comment]
seems "some unit test using etcd [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7989/pipeline/72)" is failed because etcd too slow

```
[2021-01-11T08:54:57.388Z] [2021/01/11 16:53:08.828 +08:00] [WARN] [util.go:163] ["apply request took too long"] [took=1.630559395s] [expected-duration=100ms] [prefix=] [request="header:<ID:14726057132519571993 > txn:<compare:<target:CREATE key:\"unit-test/election-evict-leader/4c5d76f0a5efc205\" create_revision:0 > success:<request_put:<key:\"unit-test/election-evict-leader/4c5d76f0a5efc205\" value_size:40 lease:5502685095664796165 >> failure:<request_range:<key:\"unit-test/election-evict-leader/4c5d76f0a5efc205\" > >>"] [response=size:14] []
...
[2021-01-11T08:54:57.388Z] [2021/01/11 16:53:08.830 +08:00] [INFO] [election.go:292] ["get response from election observe"] [component=election] [key=unit-test/election-evict-leader/4c5d76f0a5efc205] [value="{\"id\":\"member1\",\"addr\":\"127.0.0.1:1234\"}"]
```

putting election key spends more than 1 second, so in the test sleeping 1 second and checking the leader is not enough

**Review by leoppro — COMMENTED** [review]
LGTM

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**